### ISSUE TITLE
error using promises (all methods)

### ISSUE THREAD

**Issue by davelandry** [superleap/github-issues-label-sync#40]
**What is the current behavior?**
When using deleteLabels to delete multiple labels, the first label in the array is deleted, and then this error occurs:

```sh
Unhandled rejection TypeError: this.github.issues.deleteLabel(...).then(...).catch(...).lastly is not a function
    at GithubIssuesLabelSync.deleteLabel (/Users/Dave/Sites/d3plus-common/node_modules/github-issues-label-sync/lib/LabelSync.js:167:16)
    at /Users/Dave/Sites/d3plus-common/node_modules/github-issues-label-sync/lib/LabelSync.js:194:31
    at tryCatcher (/Users/Dave/Sites/d3plus-common/node_modules/bluebird/js/release/util.js:16:23)
    at MappingPromiseArray._promiseFulfilled (/Users/Dave/Sites/d3plus-common/node_modules/bluebird/js/release/map.js:57:38)
    at MappingPromiseArray.PromiseArray._iterate (/Users/Dave/Sites/d3plus-common/node_modules/bluebird/js/release/promise_array.js:113:31)
    at MappingPromiseArray.init (/Users/Dave/Sites/d3plus-common/node_modules/bluebird/js/release/promise_array.js:77:10)
    at new MappingPromiseArray (/Users/Dave/Sites/d3plus-common/node_modules/bluebird/js/release/map.js:25:10)
    at map (/Users/Dave/Sites/d3plus-common/node_modules/bluebird/js/release/map.js:152:12)
    at Promise.map (/Users/Dave/Sites/d3plus-common/node_modules/bluebird/js/release/map.js:156:12)
    at GithubIssuesLabelSync.deleteLabels (/Users/Dave/Sites/d3plus-common/node_modules/github-issues-label-sync/lib/LabelSync.js:193:40)
    at issueSync.getLabels.then.curr (/Users/Dave/Sites/d3plus-common/bin/env.js:42:13)
    at process._tickCallback (internal/process/next_tick.js:103:7)
```

**What is the expected behavior?**
For all labels to be deleted! 😏

**Comment by vrtxf** [comment]
Hi, thanks for letting me know I'll take a look at it tonight !

**Comment by davelandry** [comment]
Awesome! It's also happening with createLabels:

```sh
Unhandled rejection TypeError: this.github.issues.createLabel(...).then(...).catch(...).lastly is not a function
    at GithubIssuesLabelSync.createLabel (/Users/Dave/Sites/d3plus-common/node_modules/github-issues-label-sync/lib/LabelSync.js:240:16)
    at /Users/Dave/Sites/d3plus-common/node_modules/github-issues-label-sync/lib/LabelSync.js:265:31
    at tryCatcher (/Users/Dave/Sites/d3plus-common/node_modules/bluebird/js/release/util.js:16:23)
    at MappingPromiseArray._promiseFulfilled (/Users/Dave/Sites/d3plus-common/node_modules/bluebird/js/release/map.js:57:38)
    at MappingPromiseArray.PromiseArray._iterate (/Users/Dave/Sites/d3plus-common/node_modules/bluebird/js/release/promise_array.js:113:31)
    at MappingPromiseArray.init (/Users/Dave/Sites/d3plus-common/node_modules/bluebird/js/release/promise_array.js:77:10)
    at new MappingPromiseArray (/Users/Dave/Sites/d3plus-common/node_modules/bluebird/js/release/map.js:25:10)
    at map (/Users/Dave/Sites/d3plus-common/node_modules/bluebird/js/release/map.js:152:12)
    at Promise.map (/Users/Dave/Sites/d3plus-common/node_modules/bluebird/js/release/map.js:156:12)
    at GithubIssuesLabelSync.createLabels (/Users/Dave/Sites/d3plus-common/node_modules/github-issues-label-sync/lib/LabelSync.js:264:40)
    at Promise.all.then (/Users/Dave/Sites/d3plus-common/bin/env.js:47:15)
    at process._tickCallback (internal/process/next_tick.js:103:7)
```

**Comment by vrtxf** [comment]
Hi, I identified the issue as being configuration related. Basically I am using `bluebird` for promises but unless I pass it down in the config as `Promise: require('bluebird')` it will fail take the default argument.

I will fix this tonight and update the documentation. Here's a snippet in case you want to use this before the cli version is out

```
let config = {
    "github": {
        "user": "superleap",
        "repo": "esdoc-hacker-vision",
        "token": "demo",
        "options": {
            "debug": true,
            "followRedirects": false,
            "Promise": require('bluebird')
        }
    },
    "categories": [
        {
            "name": "GH Review",
            "labels": [
                {
                    "name": "accepted",
                    "color": "009800"
                },
                {
                    "name": "breaking changes",
                    "color": "a33cd6"
                },
                {
                    "name": "needs-revision",
                    "color": "e11d21"
                },
                {
                    "name": "on hold",
                    "color": "bfdadc"
                },
                {
                    "name": "review-needed",
                    "color": "fbca04"
                },
                {
                    "name": "shipped",
                    "color": "d4c5f9"
                }
            ]
        },
        {
            "name": "Resolution",
            "labels": [
                {
                    "name": "duplicate",
                    "color": "ccc"
                },
                {
                    "name": "invalid",
                    "color": "e6e6e6"
                },
                {
                    "name": "unresolved",
                    "color": "fef2c0"
                },
                {
                    "name": "wontfix",
                    "color": "fff"
                }
            ]
        },
        {
            "name": "Semver",
            "labels": [
                {
                    "name": "exempt",
                    "color": "1d76db"
                },
                {
                    "name": "major",
                    "color": "b60205"
                },
                {
                    "name": "minor",
                    "color": "fbca04"
                },
                {
                    "name": "patch",
                    "color": "0e8a16"
                },
                {
                    "name": "premajor",
                    "color": "e99695"
                }
            ]
        },
        {
            "name": "Status",
            "labels": [
                {
                    "name": "new",
                    "color": "006b75"
                },
                {
                    "name": "reverted",
                    "color": "d93f0b"
                },
                {
                    "name": "unconfirmed",
                    "color": "d4c5f9"
                }
            ]
        },
        {
            "name": "Type",
            "labels": [
                {
                    "name": "bug",
                    "color": "fc2929"
                },
                {
                    "name": "enhancement",
                    "color": "84b6eb"
                },
                {
                    "name": "feature-request",
                    "color": "c7def8"
                },
                {
                    "name": "question",
                    "color": "cc317c"
                },
                {
                    "name": "regression",
                    "color": "e11d21"
                }
            ]
        }
    ]
};

let {user, repo, token, options} = config.github;
let githubSync = new (require('github-issues-label-sync'))(options, user, repo, token);
let labels = [];

Array.from(config.categories).forEach((category) => {
    category.labels.forEach((label) => {
        label.name = `${category.name}: ${label.name}`;
        labels.push(label);
    });
});

// setup done
githubSync.deleteLabels(labels, false).then((response) => {
    // log raw response bodies
    console.log(response);
    // log created/updated labels
    console.log(githubSync.deletedLabels);
});
```

**Comment by vrtxf** [comment]
You can see it in action here: https://github.com/superleap/esdoc-hacker-vision/labels

**Comment by vrtxf** [comment]
Released as 1.19.0

I will add functional tests this weekend. Once I am done with that I can work on the CLI version that will solve this complicated configuration gimmick.

I'm open to feedback and suggestions. Let me know if there is anything you want this tool to do !

Cheers

**Comment by davelandry** [comment]
Awesome work @vrtxf! Thanks for such the speedy turnaround (and such a cool package).

...although I think you may have introduced another bug: #42

**Comment by davelandry** [comment]
Out of curiosity, what do you use bluebird for that ES6 Promises doesn't provide? Is it that `.lastly` method?

**Comment by vrtxf** [comment]
I went with bluebird because it was included in `github-node` and frankly I couldn't get my class to work with normal promises.
At the time I wrote the initial version I was just reading es6 promises. The github package was supposedly promisified already, though I am now thinking that I was reading a different documentation (lol) since stuff wasn't working for me as I expected. 
In all likelihood I will switch to the generic syntax if bluebird doesn't add anything to the stack, I'm not a fan of extra dependencies either :D
